realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1319,8 +1319,8 @@ private void checkMaxDepth(int maxDepth) {
         if (realmObject == null) {
             throw new IllegalArgumentException("Null objects cannot be copied from Realm.");
         }
-        if (!RealmObject.isValid(realmObject)) {
-            throw new IllegalArgumentException("RealmObject is not valid, so it cannot be copied.");
+        if (!(RealmObject.isManaged(realmObject) && RealmObject.isValid(realmObject))) {
```

Comment:
I think `||` could be more preferred in this _short-circuit OR_ ops, but most would prefer to see _De Morgan_ applied anyway.